usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateRouteColorAndTextContrast.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.routes.Route;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.warning.RouteColorAndTextContrastNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.Collection;
+
+/**
+ * Use case to validate that a Route color contrast with a Route text color.
+ */
+public class ValidateRouteColorAndTextContrast {
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     */
+    public ValidateRouteColorAndTextContrast(final GtfsDataRepository dataRepo,
+                                             final ValidationResultRepository resultRepo) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+    }
+
+    /**
+     * @param color an hexadecimal string representing a color
+     * @return the resulting luminance for the color
+     */
+    private double luminance(String color) {
+        int r = Integer.parseInt(color.substring(0, 2), 16);
+        int g = Integer.parseInt(color.substring(2, 4), 16);
+        int b = Integer.parseInt(color.substring(4, 6), 16);
+
+        return (0.2126 * r + 0.7152 * g + 0.0722 * b);
```

Comment:
Avoid magic numbers with no explanation - these should be static final constants with a name indicating what they are

Reply:
I added a comment giving the reference. It's the formula given by W3 here https://www.w3.org/TR/WCAG20-TECHS/G17.html#G17-procedure

Should I create static final constants anyways?

Reply:
After discussion with Fabrice, we figured out the link to the formula would be ok.